modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/rel/TableScanNodeExecutionTest.java

```diff
@@ -155,7 +157,11 @@ public TestInternalTableImpl(
                     mock(MvTableStorage.class),
                     mock(TxStateTableStorage.class),
                     replicaSvc,
-                    mock(HybridClock.class)
+                    mock(HybridClock.class),
+                    new ScheduledThreadPoolExecutor(
+                            Runtime.getRuntime().availableProcessors(),
+                            new NamedThreadFactory("internal-table-scheduled-pool", log)
```

Comment:
Well, I have to admit that proposed solution doesn't look good, despite the fact that I've pre-approved it earlier. new ScheduledThreadPoolExecutor() as a parameter looks crappy. In all that cases we need to shutdown an executors, etc.
So, let's consider other solutions, e.g. registering listener within replicaManager that will trigger reactive notification about replica appearance in the map. Will provide more details during the sync.